decidim-proposals/app/controllers/decidim/proposals/collaborative_drafts_controller.rb

```diff
@@ -32,6 +33,7 @@ def index
       end
 
       def show
+        redirect_to "/404" unless retrieve_collaborative_draft
```

Comment:
Thanks

Reply:
I've just checked the application and `raise ActionController::RoutingError` and `redirect_to "/404"` have different behaviours. What is preferable?

- `raise ActionController::RoutingError`
![routing_error](https://user-images.githubusercontent.com/40306853/50471475-13f40300-09b5-11e9-80f3-d478a006baac.png)

- `redirect_to "/404"` 
![404](https://user-images.githubusercontent.com/40306853/50471482-22421f00-09b5-11e9-8c8d-812ae8e32e6f.png)

I used `redirect_to "/404"` because I found it elsewhere:

https://github.com/decidim/decidim/blob/93053675be80f21955b8432f4f3f9e03d1700cc9/decidim-participatory_processes/app/controllers/decidim/participatory_processes/participatory_processes_controller.rb#L23

https://github.com/decidim/decidim/blob/93053675be80f21955b8432f4f3f9e03d1700cc9/decidim-conferences/app/controllers/decidim/conferences/conferences_controller.rb#L21

Reply:
We should be using `raise ActionController::RoutingError`, as it allows debugging in development and will render the 404 page in production.

Reply:
Thanks @mrcasals